### ISSUE TITLE
saml2 httputil Response returns str type, not bytes as is required by the WSGI standard

### ISSUE THREAD

**Issue by MarcDufresne** [IdentityPython/pysaml2#406]
I have a WSGI server running pysaml2 and when using `saml2.httputil.Response` sub-classes (e.g. `NotFound`) I get the following error:

```
[2017-04-11 13:59:49 -0400] [36836] [ERROR] Error handling request
Traceback (most recent call last):
  File ".tox/py36/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 182, in handle_request
    resp.write(item)
  File ".tox/py36/lib/python3.6/site-packages/gunicorn/http/wsgi.py", line 343, in write
    raise TypeError('%r is not a byte' % arg)
TypeError: 'message' is not a byte
```
`message` is the value of my response

This is the minimal code to trigger this issue:
```py
    def not_found(self, environ, start_response):
        response = saml2.httputil.NotFound()
        return response(environ, start_response)
```

I traced this error back to `saml2.httputil.Response#_response`, specifically, this part:

```py
        if isinstance(message, six.string_types):
            return [message]  # << This doesn't return `bytes`, thus causing Gunicorn (WSGI server) to fail
        elif isinstance(message, six.binary_type):
            return [message]
        else:
            return message
```

Changing the line above to 
```py
return [message.encode('utf-8')]  # returns bytes, Gunicorn is happy, request works
```
Caused my error to go away and my browser to show `message`, as expected

---

Now I feel like this is a big issue, so I am wondering if I am doing something wrong or somehow I'm the first one to stumble upon this issue.

System Info:
- Python version: python 3.6.1
- Gunicorn version: 19.7.1
- PySAML version: 4.4.0

I can provide additional details as needed.

**Comment by PJaros** [comment]
I have stumbled over this problem now but the error has been reported years ago. I'm beginning to wonder   is enough support for this library to build upon it.
Also I'm trying to learn SAML2 and being unable to run anything is kind of frustrating.

**Comment by erakli** [comment]
Possibly I fixed it in #561
